pennylane/pauli/pauli_arithmetic.py

```diff
@@ -312,6 +320,44 @@ def __truediv__(self, other):
             f"PauliWord can only be divided by numerical data. Attempting to divide by {other} of type {type(other)}"
         )
 
+    def commutes_with(self, other):
+        """Fast check if two PauliWords commute with each other"""
+        wires = set(self) & set(other)
+        if not wires:
+            return True
+        anticom_count = sum(anticom_map[self[wire]][other[wire]] for wire in wires)
+        return (anticom_count % 2) == 0
+
+    def _commutator(self, other):
+        """comm between two PauliWords, returns tuple (new_word, coeff) for faster arithmetic"""
+        # This may be helpful to developers that need a more lightweight comm between pauli words
+        # without creating PauliSentence classes
+
+        if self.commutes_with(other):
+            return PauliWord({}), 0.0
+        new_word, coeff = self._matmul(other)
+        return new_word, 2 * coeff
+
+    def commutator(self, other):
+        """comm between two PauliWords"""
```

Comment:
I think the docstrings for `commutator` are worth fleshing out into our standard full docstring format.

Reply:
Added ✅ also for PauliSentence.commutator()